src/trust_anchor.rs

```diff
@@ -1,103 +1,85 @@
+use pki_types::{CertificateDer, TrustAnchor};
+
 use crate::cert::{lenient_certificate_serial_number, Cert, EndEntityOrCa};
 use crate::der;
 use crate::error::{DerTypeId, Error};
 
-/// A trust anchor (a.k.a. root CA).
-///
-/// Traditionally, certificate verification libraries have represented trust
-/// anchors as full X.509 root certificates. However, those certificates
-/// contain a lot more data than is needed for verifying certificates. The
-/// `TrustAnchor` representation allows an application to store just the
-/// essential elements of trust anchors. The `TrustAnchor::try_from_cert_der`
-/// function allows converting X.509 certificates to to the minimized
-/// `TrustAnchor` representation, either at runtime or in a build script.
-#[derive(Debug)]
-pub struct TrustAnchor<'a> {
-    /// The value of the `subject` field of the trust anchor.
-    pub subject: &'a [u8],
-
-    /// The value of the `subjectPublicKeyInfo` field of the trust anchor.
-    pub spki: &'a [u8],
-
-    /// The value of a DER-encoded NameConstraints, containing name
-    /// constraints to apply to the trust anchor, if any.
-    pub name_constraints: Option<&'a [u8]>,
-}
+/// Interprets the given DER-encoded certificate as a `TrustAnchor`. The
+/// certificate is not validated. In particular, there is no check that the
+/// certificate is self-signed or even that the certificate has the cA basic
+/// constraint.
+pub fn extract_trust_anchor<'a>(cert: &'a CertificateDer<'a>) -> Result<TrustAnchor<'a>, Error> {
```

Comment:
I don't love exposing this as a free function, but I think coherence limits our options here (save for an extension trait, which feels like overkill).

Reply:
> (save for an extension trait, which feels like overkill).

Not suggesting this is required, just trying to vet my own understanding, but could a new-type wrapper also work?

Reply:
It could, but then would the caller always just discard the wrapper? Or do we keep it around and are `verify_for_usage()` callers required to wrap their `pki_types::TrustAnchor` in a `webpki::TrustAnchor`? The latter seems like a rather less attractive design, but just discarding the newtype wrapper makes it feel pretty pointless (compared to the free function).